byterun/array.c

```diff
@@ -42,39 +45,72 @@ CAMLexport int caml_is_double_array(value array)
   return (Tag_val(array) == Double_array_tag);
 }
 
+/* [ 'a array -> int -> 'a ] where 'a != float */
```

Comment:
I think this comment is wrong when the float array hack is disabled: in that case, `'a` can be `float`.  (At least one other occurrence of this below.)

Reply:
These comments are intended to be true in both cases, as a guide to write C code that works with and without the option.

Reply:
Could you perhaps clarify the comments along those lines?  I think it's potentially misleading at the moment.